### DISCUSSION TITLE
Avoid serialization of empty field

### DISCUSSION THREAD

**Discussion by KonradEichstaedt** [ksclarke/jiiify-presentation#169] [General]
Hi Kevin, 

I need to create an IIIF-Manifest Version 3.0 and used your API. The JSON Result shows always empty fields like this: 

placeholderCanvas : {"empty" : true,"present" : false }

I used the default ObjectMaper of Jackson to write the JSON String. 

How can I avoid this serialization of the empty fields? 

Many Thanks, 

Konrad

**Comment by ksclarke** [comment]
Oops, sorry, just seeing this now @KonradEichstaedt (I have fallen behind in my inbox/alerts).

This is probably because the internal (de)serializer has values set that you're not setting on your ObjectMapper:

https://github.com/ksclarke/jiiify-presentation/blob/786d7828446b0eee3dcb9dc8c681ac0fee95aa51/src/main/java/info/freelibrary/iiif/presentation/v3/utils/JSON.java#L33-L37

Just to turn the `Manifest` into a `String`, though, you should be able to use the `Manifest`'s `toString()` method (and that should use the internal mapper). The string representation that's output is JSON.

If this isn't the case for you, could you give me a little reproduceable code snippet to show the issue? Thanks!

**Comment by ksclarke** [comment]
Just to make the connection between related issues, someone else is seeing this problem too: https://github.com/ksclarke/jiiify-presentation/issues/174 -- Looking for a test case so I can understand how it's being serialized without the JSON util.

**Comment by ksclarke** [comment]
Version 0.12.3 should fix this without adding the additional module.
